### DISCUSSION TITLE
IP get blocked by newegg

### DISCUSSION THREAD

**Discussion by ryandh** [jef/streetmerchant#1048] [Help]
### IP get blocked by newegg

We're sorry, but our systems have detected the possible use of an automated program to visit Newegg.com. For more information on appropriate use of the Newegg website, please see the excerpt from Newegg.com's Policy & Agreement below. If you feel you've reached this page in error, please send an email to webmaster@newegg.com with your IP address and the error code listed below.
Error Code: 3
Your IP:  
Reference Code:

**Comment by lngchn** [comment]
Yes, noticed this too. Newegg is banning IP addresses caught scalping. So is Gamestop which has been banning IP addresses for a while. These bans are permanent, meaning you need a new IP address to get these sites working again.

**Comment by jef** [comment]
> These bans are permanent, meaning you need a new IP address to get these sites working again.

I'm not sure if that is the case. I think they are temporary, but are quite long (a few days).

**Comment by huntinganswers** [comment]
ez fix .. use a vpn

**Reply by SonnyMD** [reply]
Not necessarily - I have run into the same issue on vpn; specificially NordVPN

**Reply by ryandh** [reply]
they moved all the RTX releases through a queue system.  you have to been selected, then you can buy. 
https://www.newegg.com/product-shuffle

**Comment by jef** [comment]
Also here's this: https://github.com/jef/streetmerchant/issues/1050

**Comment by huntinganswers** [comment]
> Also here's this: #1050

is it possible to adjust each establishment independently?

**Comment by aesth0x** [comment]
I was banned by newegg yesterday at 0.5 qps. I previously ran at 0.2 qps and was never banned. I followed the instruction and sent an email to newegg webmaster, and here is what I got:

```
herefore any usage of auto refresher / loaders / crawlers may result in a permanent IP ban from Newegg.
Thanks for reaching us and we will review and release your IP address this time if we see improvement from this IP.
```

The ban on my IP was lifted after ~ 12 hours. I'm now running at 0.2 qps with caution.

**Comment by tonezz9** [comment]
I got the same message. If I download a VPN to run the bot, do I still have to get my main IP address unbanned? Will a VPN solve the problem?

**Comment by ryandh** [comment]
Newegg is done pretty much. even on VPN.

**Comment by tonezz9** [comment]
> Newegg is done pretty much. even on VPN.

What do you mean by done?

**Comment by ryandh** [comment]
> > Newegg is done pretty much. even on VPN.
> 
> What do you mean by done?

I used the VPN unlimited.  it stuck in the Recaptcha page asking "are you a human"

after a couple of hours, the IP was blocked. 

then have to choose another IP, the same thing happened again.

**Comment by ryandh** [comment]
> > Newegg is done pretty much. even on VPN.
> 
> What do you mean by done?

![image](https://user-images.githubusercontent.com/5561911/100772643-b9d7f580-33b4-11eb-8aa8-6d5e12fbd071.png)

**Reply by cuDNN-error** [reply]
Well that happened to me while I was testing my bot at 0.2 qps and running proton VPN to a us server

**Comment by tonezz9** [comment]
> > > Newegg is done pretty much. even on VPN.
> > 
> > 
> > What do you mean by done?
> 
> ![image](https://user-images.githubusercontent.com/5561911/100772643-b9d7f580-33b4-11eb-8aa8-6d5e12fbd071.png)

That's just great, so is it back to manually waiting for a drop? and do you know if Best Buy is still possible with a bot?

**Comment by huntinganswers** [comment]
boohoo.  dont use newegg then.

**Comment by thefl0yd** [comment]
can confirm; seems newegg is a dead end. It might be related to the switch to API calls. Any chance we can have a config option to revert back to original newegg page scraping?

**Comment by sortinousn** [comment]
You have to use a VPN, the IP block seems to reset daily. You'll have to keep switching VPN's or generating a new IP address daily.

**Reply by cuDNN-error** [reply]
Proxies?

**Comment by thefl0yd** [comment]
that's all well and good but even with very conservative settings newegg is blocking IPs in under an hour.

**Comment by perfectly-preserved-pie** [comment]
Yep, I had configured newegg to use a pause of 30000 and I _still_ got IP banned from Newegg.
I've decided to just use other Telegram/Discord/Twitch streams for Newegg.

**Comment by jef** [comment]
> Yeah newegg is just dead. Pinging the API might fix it but not entirely sure how to do that

I've currently disabled this because of some regressions with some other sites, but I'll be enabling it again.

Not everything on the newegg model has `itemNumber`s either, which is required to hit the API. So if those get filled out, it should work properly.

**Comment by rdfurman** [comment]
I just hit this ban as well. Running with page sleep set at 60-120 seconds.

**Comment by Z474AL3MN2VC** [comment]
The default page check is fast enough for bot detection so I set check minimums to one minute and maximums to five minutes, NewEgg still blocked my IP. I can get to the site if I use a proxy or my mobile phone but still very annoying. I don't want to increase the page times either as I am already receiving less notifications than a friend using SM. Seems NewEgg needs some special attention or lower speeds to continue working.

I'm going to temporarily remove NewEgg as a store to check due to this and hope I get unblocked. Then I'll add it later.

**Comment by ryandh** [comment]
not big deal, they moved all purchase through Newegg Shuffle only
